Description: Handle EXTRA_* flags into icedtea
 - Use -J-Xbootclasspath for javac of PulseAudio
 - Append $EXTRA_CFLAGS_IT to $IT_CFLAGS (use for PulseAudio native build)
 - Append $EXTRA_LDFLAGS_IT to PulseAudio native link
 - Send all EXTRA_* flags to jamvm and cacao
 - Remove `-icedtea' from the release identifier
 - Rename hotspot source and debian patches for zerovm build
Author: Damien Raude-Morvan <drazzib@debian.org>, Matthias Klose <doko@ubuntu.com>
 
--- Makefile.am.orig
+++ Makefile.am
@@ -133,7 +133,7 @@
 IT_JAVACFLAGS=$(IT_JAVAC_SETTINGS) -source $(IT_LANGUAGE_SOURCE_VERSION) -target $(IT_CLASS_TARGET_VERSION)
 
 # Flags
-IT_CFLAGS=$(CFLAGS) $(ARCHFLAG)
+IT_CFLAGS=$(CFLAGS) $(ARCHFLAG) $(EXTRA_CFLAGS_IT)
 
 # Conditional defintions
 
@@ -1477,9 +1477,6 @@
 	  echo "DISTRO_PACKAGE_VERSION=$(PKGVERSION)" \
 	    >>openjdk/jdk/make/common/shared/Defs.gmk ;
 endif
-	if test x"$(PROJECT_NAME)" != "xjdk7" && test x"$(PROJECT_NAME)" != "xicedtea"; then \
-	  proj_suffix="-$(PROJECT_NAME)"; \
-	fi ; \
 	if test x"$(VERSION_SUFFIX)" != "x"; then \
 	  ver_suffix="-$(VERSION_SUFFIX)"; \
 	fi ; \
@@ -1945,7 +1942,7 @@
 	 -I$(PULSE_JAVA_NATIVE_BUILDDIR) -o $@ -c $<
 
 $(PULSE_JAVA_NATIVE_BUILDDIR)/libpulse-java.so: $(PULSE_JAVA_NATIVE_OBJECTS)
-	$(CC) $(LDFLAGS) -shared $(PULSE_JAVA_NATIVE_OBJECTS) $(LIBPULSE_LIBS) \
+	$(CC) $(LDFLAGS) $(EXTRA_LDFLAGS_IT) -shared $(PULSE_JAVA_NATIVE_OBJECTS) $(LIBPULSE_LIBS) \
 	 -o $(PULSE_JAVA_NATIVE_BUILDDIR)/libpulse-java.so
 
 endif
@@ -2013,7 +2010,8 @@
 if BUILD_JAMVM
 	cd jamvm/jamvm && \
 	./autogen.sh --with-java-runtime-library=openjdk7 \
-	  --prefix=$(abs_top_builddir)/jamvm/install ; \
+	  --prefix=$(abs_top_builddir)/jamvm/install \
+	  CFLAGS='$(EXTRA_CFLAGS_JAMVM)' LDFLAGS='$(EXTRA_LDFLAGS_JAMVM)' CPPFLAGS='$(EXTRA_CPPFLAGS_JAMVM)' CXXFLAGS='$(EXTRA_CXXFLAGS_JAMVM)'; \
 	$(MAKE) ; \
 	$(MAKE) install
 	mkdir -p $(abs_top_builddir)/jamvm/install/hotspot/jre/lib/$(INSTALL_ARCH_DIR)/server
@@ -2082,7 +2080,8 @@
 	  --with-java-runtime-library=openjdk7 \
 	  --with-java-runtime-library-prefix=$(abs_top_builddir)/openjdk \
 	  --with-java-runtime-library-classes=$(RUNTIME) \
-	  --enable-jre-layout $(CACAO_CONFIGURE_ARGS); \
+	  --enable-jre-layout $(CACAO_CONFIGURE_ARGS) \
+	  CFLAGS='$(EXTRA_CFLAGS_CACAO)' LDFLAGS='$(EXTRA_LDFLAGS_CACAO)' CPPFLAGS='$(EXTRA_CPPFLAGS_CACAO)' CXXFLAGS='$(EXTRA_CXXFLAGS_CACAO)'; \
 	$(ARCH_PREFIX) $(MAKE) -j$(PARALLEL_JOBS) install
 	ln -sf server $(abs_top_builddir)/cacao/install/jre/lib/$(INSTALL_ARCH_DIR)/client
 	touch $(abs_top_builddir)/cacao/install/jre/lib/$(INSTALL_ARCH_DIR)/server/Xusage.txt
@@ -2158,7 +2157,7 @@
 			'--with-javac=% '--with-javac=% \
 			'--with-rmic=% '--with-additional-vms=% \
 			'--disable-bootstrap% '--enable-bootstrap% , \
-		$(CONFIGURE_ARGS)) \
+		$(subst hotspot-default.tar,hotspot-zero.tar,$(CONFIGURE_ARGS))) \
 	$(foreach i, openjdk hotspot corba jaxp jaxws jdk langtools, \
 	  $(if $(findstring --with-$(i)-src-zip=, $(CONFIGURE_ARGS)),, --with-$(i)-src-zip=$(abs_top_builddir)/$(i).tar.gz))
 
@@ -2168,7 +2167,7 @@
 	BUILD_JAXWS=false     ALT_JAXWS_DIST=$(BUILD_OUTPUT_DIR)/jaxws/dist \
 	BUILD_CORBA=false     ALT_CORBA_DIST=$(BUILD_OUTPUT_DIR)/corba/dist \
 	BUILD_JDK=false \
-	DISTRIBUTION_PATCHES='$(foreach p,$(DISTRIBUTION_PATCHES),$(if $(findstring cacao,$(p)),,$(if $(findstring jamvm,$(p)),,$(p))))'
+	DISTRIBUTION_PATCHES='$(foreach p,$(subst -default,-zero,$(DISTRIBUTION_PATCHES)),$(if $(findstring cacao,$(p)),,$(if $(findstring jamvm,$(p)),,$(p))))'
 
 stamps/add-zero.stamp: stamps/icedtea.stamp
 	mkdir -p stamps
--- rewriter/com/redhat/rewriter/ClassRewriter.java
+++ rewriter/com/redhat/rewriter/ClassRewriter.java
@@ -60,8 +60,8 @@
   /**
    * The executor for submitting rewriting jobs.
    */
-  private static final ExecutorService executor = DEBUG ?
-    Executors.newSingleThreadExecutor() : Executors.newCachedThreadPool();
+  private static final ExecutorService executor = 
+    Executors.newSingleThreadExecutor();
 
   /**
    * The source directory, set once by main.
diff -r 4333294206f6 hotspot.map
--- hotspot.map	Thu Feb 21 06:29:57 2013 +1100
+++ hotspot.map	Wed Feb 20 23:29:51 2013 +0100
@@ -1,3 +1,3 @@
 # version url changeset sha256sum
 default http://icedtea.classpath.org/hg/release/icedtea7-forest-2.3/hotspot 104e2c65892d 203acf2f9737032fe4bb1096e2f81417fa8a92a7016d8220b54a70efd579711a
-zero http://icedtea.classpath.org/hg/release/icedtea7-forest-2.1/hotspot 32569b4d36f4 455be170dcea6edbc9c74f9d67308bb6a1f39dadda0267e7d73ea6af3043f60c
+zero http://icedtea.classpath.org/hg/release/icedtea7-forest-2.1/hotspot d8b22e079abe 28883ebefcfff1ba8b356a7fabf8e449c25fe7fffe1d563f1f58f887100063f0
